torch/_functorch/aot_autograd.py

```diff
@@ -4306,12 +4309,25 @@ def convert(idx, x):
                         return x
                 # TODO: Ensure that this codepath is never exercised from
                 # Dynamo
+
+                def from_fake(x, static_shapes):
+                    ftensor = fake_mode.from_tensor(x, static_shapes=static_shapes, force_fresh=False)
+                    # The tensor has changed during initial trace. If we were to use the
+                    # cached memoized tensor, we would get incorrect data.
+                    # TODO(voz): We should check more than just shape... stride, dtype, etc.
+                    if ftensor.shape != x.shape:
+                        return fake_mode.from_tensor(x, static_shapes=static_shapes, force_fresh=True)
+                    return ftensor
```

Comment:
It seems like however we end up wanting to support this, it should be an implementation detail of FakeTensors cache management and not something that bleeds into AOTAutograd

Reply:
100% - this is just a hack. This also has implications in symbolic shape that I dislike (as per my comment below!) 

The new plan here is to:

1) Snapshot shape and stride at fake tensor creation time
2) Pipe those in alongside the real tensor as `from_tensor` time
3) In shape_env, use the values passed in at (2) to create a new fake_tensor with properly mapped sizes and strides as they were originally at creation time (1)